Create a nested 'kgames' menu

keith-packard opened the pull request:
This adds .menu and .directory files so that all of the Kgames applications land in a Kgames sub-menu of the Games menu. Note that the default 'ninja install' does this for just the current user as there is no directory in /usr/local which can hold .menu files, the only global directory is /etc/xdg.

keith-packard committed:
Create a nested 'kgames' menu for the desktop

This places all of the kgames applications inside a 'KGames' menu
which lives in the main Games menu.

This adds a configuration option as to whether to install the .menu
file in the current user's home directory -- there isn't a location in
/usr/local which is defined for this file, only /etc or $HOME. This is
enabled by default so that a 'ninja install' works and doesn't require
admin privs. Switch back to the /etc location by doing

	$ meson -Duser-menu=false

Signed-off-by: Keith Packard <keithp@keithp.com>

keith-packard committed:
Xkw: show "empty stack" in messages for any virtual values

Suits > spade (back, empty, none) or ranks > king (empty) don't show
'??? of ???s' anymore.

Signed-off-by: Keith Packard <keithp@keithp.com>

BartMassey commented:
The first patch obviously doesn't belong. I'll allow it.

More troublingly

    meson.build:55:0: ERROR: Tried to access unknown option "menudir".

keith-packard committed:
Update menus during install

Add a script that checks for both $DISPLAY being set and
xdg-desktop-menu being available before running that with the
'forceupdate' flag to get the current user's menu updated.

Signed-off-by: Keith Packard <keithp@keithp.com>

BartMassey commented:
Works for me!
